Import from urllib and not url

tebanep opened the pull request:
In Python 3, the correct way to import
urlparse is through urllib.parse.

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Import from urllib and not url, as that is the correct name for the module in Python3.
Currently, an ImportError is being raised by the lxd module when used with Python 3.

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
- LXD module with Python 3

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.3.0.0
  config file = 
  configured module search path = Default w/o overrides
  python version = 3.5.2 (default, Nov 17 2016, 17:05:23) [GCC 5.4.0 20160609]
```


##### ADDITIONAL INFORMATION
- The traceback is:
```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible___idoz8m/ansible_modlib.zip/ansible/module_utils/lxd.py", line 56, in <module>
ImportError: No module named 'urlparse'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/ansible___idoz8m/ansible_module_lxd_container.py", line 254, in <module>
    from ansible.module_utils.lxd import LXDClient, LXDClientException
  File "/tmp/ansible___idoz8m/ansible_modlib.zip/ansible/module_utils/lxd.py", line 59, in <module>
ImportError: No module named 'url'

failed: [10.118.6.204] (item=kinto) => {
    "failed": true,
    "item": "kinto",
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible___idoz8m/ansible_modlib.zip/ansible/module_utils/lxd.py\", line 56, in <module>\nImportError: No module named 'urlparse'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/tmp/ansible___idoz8m/ansible_module_lxd_container.py\", line 254, in <module>\n    from ansible.module_utils.lxd import LXDClient, LXDClientException\n  File \"/tmp/ansible___idoz8m/ansible_modlib.zip/ansible/module_utils/lxd.py\", line 59, in <module>\nImportError: No module named 'url'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 0
}
```

Commit:
Import from urllib and not url

In Python 3, the correct way to import
urlparse is through urllib.parse.

sivel commented:
This should preferably be updated to import from six rather than doing a try/except.  See other modules for an example.

tebanep commented:
@sivel The only thing I did was adding "lib" to the import name. The try, except clauses where already made by module's author. I don't have much experience using six, but with this change the module will run for the moment. Don't you think this can be a temporary solution?

Commit:
Use six module to import urlparse

Import urlparse from ansible.module_utils.six.
This way, the import statement is compatible both
with Python 3 and Python 2.

Commit:
Fix urlparse import using six module

The correct import is:
from six.moves.urllib.parse import urlparse

Commit:
Import six from ansible.compat

Commit:
Import from six.moves. Don't make six global.

Prevent code smell by not making six available
in the global namespace and instead importing from
one of its subpackages.

tebanep commented:
Hi @sivel, I made the proposed change with six. There are other places where six can be applied as well. Should I proceed and make them in this pull request or should I create another one? There are styling fixes I would like to do as well to be compatible with pep8 -> [pycodestyle](https://pycodestyle.readthedocs.io/en/latest/) in another branch. Is this something encouraged in Ansible?

sivel commented:
Style fixes should be done in their own PR.  This PR is sufficient as is.

tebanep commented:
Any updates regarding this PR? Can this be merged?

tebanep commented:
Or is there anything else I should do to make the PR conformant @sivel ?

tebanep commented:
Thank you!
